File: cpp/459.cpp
class Solution {
public:
    bool repeatedSubstringPattern(string s) {
        int size = s.size();
        for (int i = 0; i < size / 2; ++i) {
            string sub = s.substr(0, i + 1);
            int subSize = sub.size();
            if (size % subSize != 0)
                continue;
            int j = subSize;
            for (; j < size; j += subSize) {
                if (sub != s.substr(j, subSize))
                    break;
            }
            if (j == size && s.substr(j-subSize, subSize) == sub)
                return true;
        }
        return false;
    }
};

File: go/459.go
func repeatedSubstringPattern(s string) bool {
    size := len(s)
    for i := 1; i <= size/2; i++ {
        sub := s[0:i]
        if size%i == 0 {
            temp := ""
            for j := 0; j < size; j += i {
                temp += sub
            }
            if temp == s {
                return true
            }
        }
    }
    return false
}

func repeatedSubstringPattern1(s string) bool {
    size := len(s)
    for i := 1; i <= size/2; i++ {
        sub := s[0:i]
        if size%i == 0 {
            j := i
            for ; j < size; j += i {
                if s[j:i+j] != sub {
                    break
                }
            }
            if j == size && s[j-i:] == sub {
                return true
            }
        }
    }
    return false
}

